packages/firestore/src/core/firestore_client.ts

```diff
@@ -495,10 +495,10 @@ export class FirestoreClient {
   }
 
   shutdown(): Promise<void> {
-    if (this.isShutdown === true) {
-      return Promise.resolve();
-    }
     return this.asyncQueue.enqueue(async () => {
+      if (this.isShutdown === true) {
+        return Promise.resolve();
```

Comment:
nit: This is already inside an async function returning `Promise<void>`, so can't you just `return;`?

Also on Android you made it such that the `if` tests the equivalent `!this.isShutdown`. This is different.